pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PublicSettings.scala

```diff
@@ -91,56 +59,17 @@ object PublicSettings {
    * @param client implicit dispatch.Http to use for fetching the key
    * @param ec     implicit execution context to use for fetching the key
    */
-  def getPublicKey(domain: String, bucketName: String)(implicit client: OkHttpClient, ec: ExecutionContext): Future[PublicKey] = {
-    getPublicSettings(domain, bucketName) flatMap extractPublicKey
+  def getPublicKey(settingsFileKey: String, bucketName: String, s3Client: AmazonS3): Either[SettingsFailure, PublicKey] = {
+    fetchSettings(settingsFileKey, bucketName, s3Client) flatMap extractSettings flatMap extractPublicKey
   }
 
-  // internal functions for fetching and parsing the responses
-  private def fetchSettings(domain: String, bucketName: String)(implicit client: OkHttpClient, ec: ExecutionContext): Future[Either[Throwable, String]] = {
-    val promise = Promise[Either[Throwable, String]]()
-    val req = new Request.Builder().url(s"https://s3-eu-west-1.amazonaws.com/$bucketName/$domain.settings.public").build()
+  private[pandomainauth] def extractPublicKey(settings: Map[String, String]): Either[SettingsFailure, PublicKey] = for {
+    rawKey <- settings.get("publicKey").toRight(PublicKeyNotFoundFailure).right
+    publicKey <- validateKey(PublicKey(rawKey)).right
+  } yield publicKey
 
-    client.newCall(req).enqueue(new Callback {
-      override def onFailure(call: Call, e: IOException): Unit = promise.success(Left(e))
-      override def onResponse(call: Call, response: Response): Unit = promise.success(Right(response.body.string))
-    })
-
-    promise.future
-  }
-
-  private[pandomainauth] def extractSettings(settingsAttempt: Either[Throwable, String]): Future[Map[String, String]] =
```

Comment:
These methods have been moved to the `Settings` object so they can be shared with the other refresher